operator/src/lib.rs

```diff
@@ -75,37 +76,142 @@ impl NifiState {
+            config,
+            &zookeeper_info.connection_string,
+            node_name,
+        );
+        let state_management_xml = create_state_management_xml(
+            &self.context.resource.spec,
+            &zookeeper_info.connection_string,
+        );
+
+        let mut data = BTreeMap::new();
+        data.insert("bootstrap.conf".to_string(), bootstrap_conf);
+        data.insert("nifi.properties".to_string(), nifi_properties);
+        data.insert("state-management.xml".to_string(), state_management_xml);
+
         match self
             .context
             .client
-            .get::<ConfigMap>(name, Some(&"default".to_string()))
+            .get::<ConfigMap>(cm_name, Some(&"default".to_string()))
             .await
         {
-            // TODO: check and compare content here
-            Ok(_) => {
-                debug!("ConfigMap [{}] already exists, skipping creation!", name);
-                return Ok(());
+            Ok(config_map) => {
+                if let Some(existing_config_map_data) = config_map.data {
+                    if existing_config_map_data == data {
+                        debug!(
+                            "ConfigMap [{}] already exists with identical data, skipping creation!",
+                            cm_name
+                        );
+                        return Ok(());
+                    } else {
+                        debug!(
+                            "ConfigMap [{}] already exists, but differs, recreating it!",
+                            cm_name
+                        );
+                    }
+                }
             }
             Err(e) => {
                 // TODO: This is shit, but works for now. If there is an actual error in comes with
                 //   K8S, it will most probably also occur further down and be properly handled
-                debug!("Error getting ConfigMap [{}]: [{:?}]", name, e);
+                debug!("Error getting ConfigMap [{}]: [{:?}]", cm_name, e);
             }
         }
 
-        let config = create_config_file(config);
-
-        let mut data = BTreeMap::new();
-        data.insert("nifi.properties".to_string(), config);
-
-        // And now create the actual ConfigMap
-        let cm =
-            stackable_operator::config_map::create_config_map(&self.context.resource, &name, data)?;
+        let cm = stackable_operator::config_map::create_config_map(
+            &self.context.resource,
+            &cm_name,
+            data,
+        )?;
         self.context.client.create(&cm).await?;
         Ok(())
     }
 
+    async fn wait_for_host_name_and_create_config_map(&self) -> NifiReconcileResult {
+        for pod in &self.existing_pods {
+            let pod_name = match &pod.metadata.name {
+                Some(name) => name,
+                None => {
+                    error!("Pod has no name set. This should not happen!");
+                    return Ok(ReconcileFunctionAction::Done);
```

Comment:
Should this not be a Requeue? Or delete? Come to think of it this really should not ever happen because Kubernetes wouldn't allow it.

Reply:
Yeah i thought about a loop continue...it actually really cant happen.